Update self-healing.md (#4551)
MicrosoftDocs/architecture-center: docs/guide/design-principles/self-healing.md

## Recommendations

> **Use decoupled components that communicate asynchronously**. Ideally, components are decoupled in terms of time and space. Decoupled in time means that components do not need to be present at the same time for communication to be possible. Decoupled in space means that the sender and receiver do not have to run in the same process but wherever it’s more efficient. Decoupled components ideally use events to communicate with each other. This helps to minimize the chance of cascading failures.

Reviewer: ```suggestion
**Use decoupled components that communicate asynchronously**. Ideally, components are decoupled in terms of time and space. Decoupled in time means that components don't need to be present at the same time for communication to be possible. Decoupled in space means that the sender and receiver don't have to run in the same process, but they can wherever it's more efficient. Decoupled components ideally use events to communicate with each other. This helps minimize the chance of cascading failures.
```
Minor edits

> **Use decoupled components that communicate asynchronously**. Note that coupling is a multi-dimensional continuum. Ideally, components should be decoupled in terms of time and space. Decoupled in time means that components do not need to be present at the same time for communication to be possible. Decoupled in space means that the sender and receiver do not have to run in the same process but wherever it’s more efficient.  Decoupled components should ideally use events to communicate with each other. This helps to minimize the chance of cascading failures.

Reviewer 2: ```suggestion
**Use decoupled components that communicate asynchronously**. Ideally, components are decoupled in terms of time and space. Decoupled in time means that components do not need to be present at the same time for communication to be possible. Decoupled in space means that the sender and receiver do not have to run in the same process but wherever it’s more efficient. Decoupled components ideally use events to communicate with each other. This helps to minimize the chance of cascading failures.
```
